internal/tenant/tenanttest/tenant_test.go

```diff
@@ -33,100 +25,82 @@ func TestTenantWithPluginInstanceManagement(t *testing.T) {
 	factoryInvocations := 0
 	factory := datasource.InstanceFactoryFunc(func(settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
 		factoryInvocations++
-		return &testPlugin{
-			settings: dataSourceSettings{},
-		}, nil
+		return &testPlugin{}, nil
 	})
-	instancePrvdr := datasource.NewInstanceProvider(factory)
-	instanceMgr := instancemgmt.New(instancePrvdr)
-	handler := automanagement.NewManager(instanceMgr)
-
-	pCtx := backend.PluginContext{DataSourceInstanceSettings: &backend.DataSourceInstanceSettings{}}
-	qdr := &backend.QueryDataRequest{PluginContext: pCtx}
-	crr := &backend.CallResourceRequest{PluginContext: pCtx}
-	chr := &backend.CheckHealthRequest{PluginContext: pCtx}
-	responseSender := newTestCallResourceResponseSender()
-
-	go func() {
-		err := backend.GracefulStandaloneServe(backend.ServeOpts{
-			QueryDataHandler:    handler,
-			CallResourceHandler: handler,
-			StreamHandler:       handler,
-			CheckHealthHandler:  handler,
-		}, standalone.NewServerSettings(addr, t.TempDir()))
-		require.NoError(t, err)
-	}()
-
-	pc, shutdown, err := newPluginClient(addr)
+	tp, err := experimentalDS.Manage(factory, experimentalDS.ManageOpts{Address: addr})
 	require.NoError(t, err)
 	defer func() {
-		err = shutdown()
-		require.NoError(t, err)
+		err = tp.Client.Shutdown()
+		t.Log("client shutdown error", err)
+		err = tp.Server.Shutdown()
+		t.Log("server shutdown error", err)
```

Comment:
I would consider adding a `Shutdown` method to `TestPlugin` to make it easier for whoever is writing the tests so they don't forget that they have to call `Shutdown` both on client and server. Something like:

```go
func (tp *TestPlugin) Shutdown() error {
    clErr := tp.Client.Shutdown()
    srvErr := tp.Server.Shutdown()
    return errors.Join(clErr, srvErr)
}
```

and then in the test it becomes shorter:

```go
defer func() {
    require.NoError(t, tp.Shutdown())
}()
```

WDYT?

Reply:
Bellissimo! ❤️